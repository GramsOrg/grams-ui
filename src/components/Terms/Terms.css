.scrollable {
    overflow-y: scroll;
    margin-bottom: 1em;
}

.small {
    height: 200px;
}

.medium {
    height: 400px;
}

.large {
    height: 600px;
}
