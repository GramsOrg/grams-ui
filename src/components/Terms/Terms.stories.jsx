import React from 'react';

import { action } from "@storybook/addon-actions";
import Content from './Terms.md';

import Terms from './Terms.tsx';

export default {
  title: 'Onboarding/Terms',
  component: Terms,
  parameters: {
    componentSubtitle: 'Displays and prompts the user to agree to the terms of use',
    storyshots: { disable: false },
  },
  argTypes: {
    dir: {
      control: 'radio',
      options: ['ltr', 'rtl']
    },
    inverted: {
      control: 'boolean'
    }
  }
};

const Template = ({dir, inverted, ...props}) => {
  return (
    <Terms
      dir={dir}
      inverted={inverted}
      onAccept={action("Trigger onAccept()")}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  dir: 'ltr',
  inverted: false
}

export const Populated = Template.bind({});
Populated.args = {
  dir: 'ltr',
  content: Content,
  inverted: false
}

export const RTL = Template.bind({});
RTL.args = {
  dir: 'rtl',
  content: Content,
  inverted: false
}
