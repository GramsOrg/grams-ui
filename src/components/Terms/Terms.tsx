import React, { useState } from 'react';

import { useTranslation } from 'react-i18next';

import { useDirection } from '../../';

import ReactMarkdown from 'react-markdown';

import { Button, Checkbox, Segment } from 'semantic-ui-react';

import './Terms.css';

export interface ITermsProps {

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /**
   * The markdown content to be rendered as the terms of use.
   */
  content: string;

  /**
   * The size of the terms container.
   */
  size?: 'small' | 'medium' | 'large';

  /**
   * Whether the checkbox should be checked by default.
   */
  defaultChecked?: boolean;

  /**
   * The callback function to be called when the user accepts the acceptance of the terms.
   */
  onAccept?: () => void;
}

const defaultProps = {
  inverted: false,
  size: 'medium',
  defaultChecked: false
};

const Terms = (props: ITermsProps) => {

  const { t } = useTranslation();
  const direction = useDirection();

  const [accepted, setAccepted] = useState<boolean>(props.defaultChecked || false);

  const markdown: string = props.content || '';

  return (
    <div dir={direction}>
      <ReactMarkdown className={`scrollable ${props.size}`}>
        {markdown}
      </ReactMarkdown>
      <Segment basic>
        <Checkbox
          label={t('Terms.checkbox')}
          checked={accepted}
          onChange={(e, { checked }) => setAccepted(!!checked)}
        />
      </Segment>
      <Button
        primary
        fluid
        dir={direction}
        disabled={!accepted}
        onClick={() => props?.onAccept?.()}
      >
        {t('Terms.confirm')}
      </Button>
    </div>
  );
};

Terms.defaultProps = defaultProps;

export default Terms;
