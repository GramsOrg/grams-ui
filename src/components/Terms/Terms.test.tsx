/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import Terms from './Terms';

describe('Terms Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<Terms />);
    expect(asFragment()).toMatchSnapshot();
  });
});
