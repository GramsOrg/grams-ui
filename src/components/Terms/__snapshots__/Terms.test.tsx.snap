// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terms Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    dir="ltr"
  >
    <div
      class="scrollable medium"
    />
    <div
      class="ui basic segment"
    >
      <div
        class="ui checkbox"
      >
        <input
          class="hidden"
          readonly=""
          tabindex="0"
          type="checkbox"
          value=""
        />
        <label>
          I accept the terms and conditions
        </label>
      </div>
    </div>
    <button
      class="ui fluid primary disabled button"
      dir="ltr"
      disabled=""
      tabindex="-1"
    >
      Accept
    </button>
  </div>
</DocumentFragment>
`;
